packages/cutler/bin/cutler.dart

```diff
@@ -0,0 +1,279 @@
+    }
+    return output.split('\n');
+  }
+
+  Version getForkPoint() {
+    // Could use --fork-point?
+    // https://git-scm.com/docs/git-merge-base
+    final hash = runCommand(
+      "git",
+      ["merge-base", "--fork-point", upstreamBranch, releaseBranch],
+      workingDirectory: _workingDirectory,
+    );
+    return versionFrom(hash);
+  }
+
+  String contentsAtPath(String commit, String path) {
+    return runCommand("git", ["show", "$commit:$path"],
+        workingDirectory: _workingDirectory);
+  }
+
+  void writeFile(String path, String contents) {
+    File(path).writeAsStringSync(contents);
+  }
+
+  void commit(String message) {
+    runCommand("git", ["commit", "-a", "-m", message],
+        workingDirectory: _workingDirectory);
+  }
+
+  Version localHead() {
+    return versionFrom(runCommand("git", ["rev-parse", "HEAD"],
+        workingDirectory: _workingDirectory));
+  }
+}
+
+extension VersionCommands on Version {
+  String contentsAtPath(String path) {
+    return repo.contentsAtPath(hash, path);
+  }
+}
+
+String printLatestForBranch(Repo repo, String branch) {
+  final hash = repo.getLatestCommit(branch);
+  final tags = repo.getTagsFor(hash);
+  final tagsString = tags.isEmpty ? "" : " (${tags.join(', ')})";
+  print("${repo.name.padRight(10)} ${branch.padRight(25)} $hash$tagsString");
+  return hash;
+}
+
+void printVersions(VersionSet versions, int indent) {
+  print("${' ' * indent}engine    ${versions.engine}");
+  print("${' ' * indent}flutter   ${versions.flutter}");
+  print("${' ' * indent}buildroot ${versions.buildroot}");
+}
+
+VersionSet getFlutterVersions(String flutterHash) {
+  final engineHash = Repo.flutter
+      .contentsAtPath(flutterHash, 'bin/internal/engine.version')
+      .trim();
+  final depsContents =
+      Repo.engine.contentsAtPath(engineHash, Paths.engineDEPS.path);
+  final buildrootVersion = parseBuildRoot(depsContents);
+  return VersionSet(
+    engine: Repo.engine.versionFrom(engineHash),
+    flutter: Repo.flutter.versionFrom(flutterHash),
+    buildroot: Repo.buildroot.versionFrom(buildrootVersion),
+  );
+}
+
+String parseBuildRoot(String depsContents) {
+  final lines = depsContents.split('\n');
+  // Example:
+  //   'src': 'https://github.com/flutter/buildroot.git' + '@' + '059d155b4d452efd9c4427c45cddfd9445144869',
+  final buildrootLine = lines.firstWhere((line) => line.contains("'src': "));
+  final regexp = RegExp(r'([0-9a-f]{40})');
+  final match = regexp.firstMatch(buildrootLine);
+  if (match == null) {
+    throw Exception("Failed to parse buildroot version from $buildrootLine");
+  }
+  return match.group(0)!;
+}
+
+VersionSet getForkpoints() {
+  final flutterForkpoint = Repo.flutter.getForkPoint();
+  // final engineForkpoint = Repo.engine.getForkPoint();
+  // final buildrootForkpoint = Repo.buildroot.getForkPoint();
+  // return VersionSet(
+  //   engine: engineForkpoint,
+  //   flutter: flutterForkpoint,
+  //   buildroot: buildrootForkpoint,
+  // );
```

Comment:
Can we remove these lines if they're not needed?

Reply:
This is still somewhat wip.